ml_tools/buffercache.py

```diff
@@ -0,0 +1,67 @@
+import h5py
+import os
+import numpy as np
+from multiprocessing import Lock
+
+
+class BufferCache:
+    def __init__(self, cptv_name, keep_open=True, delete_if_exists=True):
+        basename = os.path.splitext(os.path.basename(cptv_name))[0]
+        self.filename = "cache-tmp/" + basename + ".tmp"
```

Comment:
Is it ok to use a relative path like this? Do you have any guarantees of what the working directory is when this is called. It might be best to create the temporary file in the same directory as the CPTV file. Perhaps just append `.cache` to the CPTV file name?

Reply:
sounds good